src/uwtools/drivers/driver.py

```diff
@@ -31,40 +32,62 @@ class Assets(ABC):
 
     def __init__(
         self,
-        config: Optional[Union[dict, Path]],
-        dry_run: bool = False,
-        batch: bool = False,
         cycle: Optional[datetime] = None,
         leadtime: Optional[timedelta] = None,
+        config: Optional[Union[dict, Path]] = None,
+        dry_run: bool = False,
         key_path: Optional[list[str]] = None,
     ) -> None:
         """
         A component driver.
 
-        :param config: Path to config file (read stdin if missing or None).
-        :param dry_run: Run in dry-run mode?
-        :param batch: Run component via the batch system?
         :param cycle: The cycle.
         :param leadtime: The leadtime.
+        :param config: Path to config file (read stdin if missing or None).
+        :param dry_run: Run in dry-run mode?
         :param key_path: Keys leading through the config to the driver's configuration block.
```

Comment:
I reordered the arguments lists for the `Asset` / `Driver` class hierarchy, and the calls between them via `super()`. Non-optional arguments are of course first, but the optional arguments now appear in the same order as in the `Assets` constructor, with `batch` tacked on to the end for `Driver` and its subclasses.

I also harmonized the type annotation for the `config` arguments across all these classes.